### PR TITLE
feat: optimize logic when checkConflict and bulkCreate

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by Hailong-am — APPROVED** [review]

File: src/core/server/saved_objects/service/lib/repository.ts [hunk 1]
```diff
@@ -668,24 +662,28 @@ export class SavedObjectsRepository {
...
+          const transformedObject = this._serializer.rawToSavedObject(doc as SavedObjectsRawDoc);
+          const filteredWorkspaces = SavedObjectsUtils.filterWorkspacesAccordingToBaseWorkspaces(
+            options.workspaces,
+            transformedObject.workspaces
+          );
+          if (!filteredWorkspaces.length) {
+            workspaceConflict = true;
+          }
         }
+        errors.push({
```
Comment (wanglam) [inline src/core/server/saved_objects/service/lib/repository.ts:676]: Is this means we will always return errors when document found?

File: src/core/server/saved_objects/service/lib/repository.ts [hunk 2]
```diff
@@ -289,11 +289,17 @@ export class SavedObjectsRepository {
 
     let savedObjectWorkspaces = workspaces;
```
Comment (wanglam) [inline src/core/server/saved_objects/service/lib/repository.ts:290]: Will the savedObjectWorkspaces be set to undefined if `options.workspaces` not exists?

File: src/core/server/saved_objects/service/lib/repository.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply src/core/server/saved_objects/service/lib/repository.ts:676]: Yes. for create / bulkCreate, we will always return an error when workspace checked conflict.

File: src/core/server/saved_objects/service/lib/repository.ts (hunk 2 above)
Comment (SuZhou-Joe) [reply src/core/server/saved_objects/service/lib/repository.ts]: Good catch, for such case, we need to call get when create-with-override

**Review by wanglam — APPROVED** [review]

**Pull request merged**